gridt/src/app/core/api.service.spec.ts

```diff
@@ -1,96 +1,155 @@
+  });
+
+  it('should fail to read identity from server when not logged in', () => {
+    expect(
+      service.userIdentity$
+    ).toBeObservable(cold("#", {}, "Can't authenticate: no credentials"));
+  });
+
+  it('should fail to update bio when not logged in', () => {
+    let mock_bio = "My very first bio."
+    expect(service.changeBio$( mock_bio )).toBeTruthy( cold('#', null, "Can't authenticate: no credentials"));
+  });
+
+  it('should fail to update password when not logged in', () => {
+    let old_password = "ABCDEF";
+    let new_password = "abcdef";
+
+    expect(service.changePassword$( old_password, new_password )).toBeTruthy( cold('#', null, "Can't authenticate: no credentials"));
+  });
+});
+
+describe("ApiService_succesful_auth", () => {
+
+  beforeEach(() => {
     httpClientStub = jasmine.createSpyObj(
-      'httpClient', ['get', 'post']
+      'httpClient', ['get', 'post', 'put']
     );
 
-    authServiceStub = jasmine.createSpyObj(
-      'auth', ['readyAuthentication$']
-    );
-    // All but a few tests presume a logged in state. readyAuthentication$ is
-    // defined to always be able to login and easy to compare with the correct
-    // headers. (This feature is important, as it makes or breaks communication
-    // with the server.)
-    authServiceStub.readyAuthentication$ = of(default_headers);
+    TestBed.configureTestingModule({
+      providers: [
+        ApiService,
+        {provide: AuthService, useClass: authServiceStub_succes},
+        {provide: HttpClient, useValue: httpClientStub}
+      ]
+    });
 
-    mock_movements = [
-      {
-        id: 10129312983,
-        name: "Mock Name",
-        short_description: "Mocking the short description",
-        description: "I will mock description. You are a fool, description.",
-        subscribed: false,
-        interval: "daily"
-      },
-      {
-        id: 1283912983123,
-        name: "Another mocked movement",
-        short_description: "This one only has a short description and no long one",
-        subscribed: false,
-        interval: "twice daily"
-      },
-      {
-        id: 10023123,
-        name: "Subscribed movement",
-        short_description: "This movement is supposed to come from the server.",
-        description: "We need some way to fake a response from the server, telling the client that it \
-         is subscribed to this particular movement",
-        subscribed: true,
-        interval: "daily"
-      },
-      {
-        id: 112312983,
-        name: "Flossing",
-        short_description: "Floss every day",
-        description: "We floss every day because it is good for our theeth.",
-        subscribed: true,
-        interval: "daily"
-      }
-    ];
-    mock_subscriptions = [...mock_movements].filter(m => m.subscribed);
-  });
+    service = TestBed.get(ApiService as Type<ApiService>);
 
-  afterEach(() => {
-    httpMock.verify();
+    mock_subscriptions = [...mock_movements].filter(m => m.subscribed);
```

Comment:
This is an inititialization for the next round, it is more logical to do this in `beforeEach` and remove this function.

Reply:
But it is in beforeEach right now, and indeed the afterEach() function is removed.